packages/subapp-server/lib/setup-hapi-routes.js

```diff
@@ -108,12 +108,15 @@ function setupRouteRender({ subAppsByPath, srcDir, routeOptions }) {
     routeOptions.__internals.subApps = [].concat(routeOptions.subApps).map(x => {
       let options = {};
       if (Array.isArray(x)) {
-        options = x[1];
+        options = x[1] || {};
         x = x[0];
       }
       // absolute: use as path
       // else: assume dir under srcDir
       // TBD: handle it being a module
+      if (x.indexOf("/") === -1) {
```

Comment:
what's this trying to do?

Reply:
This is to handle the scenario where subapp name is just the module name (when loading from a separate repo)

```
"/demo1": {
    pageTitle: "Welcome to Demo1",
    subApps: [["demo1", demo1.subAppOptions]],
  },
```
For the sub-apps from a separate repo, its path is set as  ```Path.dirname(require.resolve(`${modName}/package.json`))``` in `subAppsByPath`. Same logic is used here to retrive the details from the map.

Another option I was considering it is to set `module:true` in `subappOptions` and use that flag instead of checking for `/` in url. But that would be one additional thing to be set by the main app.

In existing code, if path is not absolute, we assume it's relative to app src dir.

Reply:
Added `module:true` flag to replace `(x.indexOf("/") === -1)` to handle module name being `@scope/name` also.

Reply:
Updated the code to remove `module:true` flag